### PR TITLE
Add Sku parameter for Set-AzureResource and New-AzureResource

### PR DESCRIPTION
**Pull request by chidmdxx**
This will allow to use the Sku property in the Set-AzureResource and New-AzureResource cmdlets, also it adds the ResourceSku class

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@chidmdxx__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Implementation/SetAzureResourceCmdlet.cs [hunk 1]
```diff
@@ -176,6 +185,7 @@ private JToken GetResourceBody()
                     {
                         Kind = this.Kind ?? resource.Kind,
                         Plan = this.Plan.ToDictionary(addValueLayer: false).ToJson().FromJson<ResourcePlan>() ?? resource.Plan,
+                        Sku = this.Sku.ToDictionary(addValueLayer: false).ToJson().FromJson<ResourceSku>(),
```
Comment (vivsriaus) [inline src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Implementation/SetAzureResourceCmdlet.cs]: Do we need to null coalesce the sku, similar to Plan above?

File: src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Entities/Resources/ResourceSku.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Entities.Resources
+{
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// The resource plan object.
```
Comment (vivsriaus) [inline src/ResourceManager/ResourceManager/Commands.ResourceManager/Cmdlets/Entities/Resources/ResourceSku.cs]: resource sku object

**Comment by vivsriaus** [comment]
@chidmdxx Can we please add a scenario test for these changes?

**Comment by ogail** [comment]
Same comment [here](https://github.com/Azure/azure-powershell/pull/854#issuecomment-140242380)

**Comment by ogail** [comment]
@azuresdkci add to whitelist

**Comment by chidmdxx** [comment]
used the GitHub UI and made a mess in the merge, will redo the changes

**Pull request closed without merging**